draft-ietf-trans-rfc6962-bis.md

```diff
@@ -0,0 +1,2193 @@
+algorithm used by each log is expected to be specified as part of the metadata
+relating to that log (see {{metadata}}).  We have established a registry of
+acceptable algorithms, see {{hash_algorithms}}. The hashing algorithm in use is
+referred to as HASH throughout this document and the size of its output in bytes
+as HASH_SIZE.  The input to the Merkle Tree Hash is a list of data entries;
+these entries will be hashed to form the leaves of the Merkle Hash Tree.  The
+output is a single HASH_SIZE Merkle Tree Hash.  Given an ordered list of n
+inputs, D\[n] = {d(0), d(1), ..., d(n-1)}, the Merkle Tree Hash (MTH) is thus
+defined as follows:
+
+The hash of an empty list is the hash of an empty string:
+
+~~~~~~~~~~~
+MTH({}) = HASH().
+~~~~~~~~~~~
+
+The hash of a list with one entry (also known as a leaf hash) is:
+
+~~~~~~~~~~~
+MTH({d(0)}) = HASH(0x00 || d(0)).
+~~~~~~~~~~~
+
+For n > 1, let k be the largest power of two smaller than n
+(i.e., k \< n \<= 2k).
+The Merkle Tree Hash of an n-element list D\[n] is then defined recursively as
+
+~~~~~~~~~~~
+MTH(D[n]) = HASH(0x01 || MTH(D[0:k]) || MTH(D[k:n])),
+~~~~~~~~~~~
+
+where || is concatenation and D\[k1:k2] denotes the list {d(k1), d(k1+1),...,
+d(k2-1)} of length (k2 - k1).  (Note that the hash calculations for leaves and
+nodes differ.  This domain separation is required to give second preimage
+resistance.)
+
+Note that we do not require the length of the input list to be a power of two.
+The resulting Merkle Tree may thus not be balanced; however, its shape is
+uniquely determined by the number of leaves.  (Note: This Merkle Tree is
+essentially the same as the history tree [CrosbyWallach] proposal, except our
+definition handles non-full trees differently.)
+
+### Merkle Inclusion Proofs    {#merkle_inclusion_proof}
+
+A Merkle inclusion proof for a leaf in a Merkle Hash Tree is the shortest list
+of additional nodes in the Merkle Tree required to compute the Merkle Tree Hash
+for that tree.  Each node in the tree is either a leaf node or is computed from
+the two nodes immediately below it (i.e., towards the leaves).  At each step up
+the tree (towards the root), a node from the inclusion proof is combined with
+the node computed so far.  In other words, the inclusion proof consists of the
+list of missing nodes required to compute the nodes leading from a leaf to the
+root of the tree.  If the root computed from the inclusion proof matches the
+true root, then the inclusion proof proves that the leaf exists in the tree.
+
+Given an ordered list of n inputs to the tree, D\[n] = {d(0), ..., d(n-1)}, the
+Merkle inclusion proof PATH(m, D\[n]) for the (m+1)th input d(m), 0 \<= m \< n,
+is defined as follows:
+
+The proof for the single leaf in a tree with a one-element input list D\[1] =
+{d(0)} is empty:
+
+~~~~~~~~~~~
+PATH(0, {d(0)}) = {}
+~~~~~~~~~~~
+
+For n > 1, let k be the largest power of two smaller than n.  The proof for the
```

Comment:
Hmmm, it looks like it wasn't there in the original.  I guess my converter must've converted all occurrences of `". "` to `".  "`